ompi/mca/pml/ucx/pml_ucx.c

```diff
@@ -440,8 +459,27 @@ int mca_pml_ucx_del_procs(struct ompi_proc_t **procs, size_t nprocs)
      * finalize gracefully */
     ucp_worker_flush(ompi_pml_ucx.ucp_worker);
 
+    /* Some peers may be in active communication phase and
+     * require progress from current process. Unfortunately
+     * PMIX could not provide progress call during Fence, and
+     * we will create own progress thread for UCX */
+    progress_data.worker   = ompi_pml_ucx.ucp_worker;
+    progress_data.complete = 0;
+
+    thread_err = pthread_create(&progress_thr, NULL, mca_pml_async_ucx_progress,
```

Comment:
we should aim for a solution in UCX, not OMPI. This problem can happen in other MPI's/PGAS libraries as well

Reply:
Or you could use the non-blocking PMIx _Fence_nb so PMIx can progress during the fence call.

Reply:
right, but it will require significant re-implementation of async progress in UCX, not sure if we have time for this now

Reply:
@rhc54 good idea, will try

Reply:
@yosefe done
cherry-pick it into v3.1?